tests/eosio.system_tests.cpp

```diff
@@ -2615,6 +2615,38 @@ BOOST_FIXTURE_TEST_CASE( eosioram_ramusage, eosio_system_tester ) try {
 
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE( ram_gift, eosio_system_tester ) try {
```

Comment:
I would like to see a few other things tested in the unit tests:

1. Take the difference between the RAM bytes from `get_account_limits` and the RAM bytes from the system contract table and make sure the  difference exactly equals `ram_gift_bytes`.

2. Try selling a very small amount of RAM (e.g. 1 byte), and make sure that the decrease in your RAM quota is not more than 100 bytes. This test would have failed (assuming `ram_gift_bytes == 1400` and the amount of bytes you are selling is, for example, less than 10 bytes) if you didn't adjust the RAM amount appropriately in the `set_resource_limits` call in `system_contract::sellram`.